pcsx2/DEV9/smap.cpp

```diff
@@ -153,6 +153,30 @@ void tx_process()
 			{
 				Console.Error("DEV9: SMAP: ERROR: odd , !pbd->pointer>0x1000 | 0x%X %u", pbd->pointer, pbd->length);
 			}
+
+			// SMAP drivers send a very specfic frame during init, then check SPD_R_INTR_STAT for SMAP_INTR_RXEND | SMAP_INTR_TXEND | SMAP_INTR_TXDNV.
+			// SMAP_INTR_TXEND is set normally, SMAP_INTR_RXEND is supposed to be set here, but we currently don't emulate that.
+			// SMAP_INTR_TXDNV is set somewhere, unsure where, we only set it in failure (instead of SMAP_INTR_TXEND), but is included in the hack here.
+			if (pbd->length == 0x5EA && pbd->pointer == 0x1000)
+			{
+				u32* ptr = (u32*)&dev9.txfifo[base];
+
+				bool test = true;
+				for (int i = 0; i < 0x5EA; i += 4)
+				{
+					if (ptr[i / 4] != i)
```

Comment:
You're comparing `u32` and `int` here which ends up causing a warning.
You could probably do some pointer arithmetic here instead:
ie
```
if(*ptr++ != i)
```

If you like the current way better, mark `ptr` as const :)

Reply:
Amended, also took your suggestion and switched to pointer arithmetic